lib/consumer_supervisor.ex

```diff
@@ -262,6 +258,23 @@ defmodule ConsumerSupervisor do
     call(supervisor, :count_children)
   end
 
+  @doc """
+  Receives a template to initialize and a set of options.
+
+  This is typically invoked at the end of the init/1 callback of module-based supervisors.
+
+  This function returns a the child specification and the supervisor flags.
+
+  """
+  def init([{_, _, _, _, _, _} = template], opts) do
+    {:ok, {_, spec}} = Supervisor.Spec.supervise([template], opts)
+    {:ok, spec, opts}
+  end
+  def init([template], opts) do
+    {:ok, {_, spec}} = Supervisor.init([template], opts)
```

Comment:
@josevalim Solved with 98c8963

I also added an if to the other clause & the tests to verify if `Supervisor.init/2` exists. It would be good to add Elixir 1.4 & 1.5 to the Travis Config therefore. If you're interested, I'll provide a separate PR.